spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/serviceregistry/ConsulAutoRegistration.java

```diff
@@ -166,10 +166,10 @@ public static void managementCustomize(
 	public static String getInstanceId(ConsulDiscoveryProperties properties,
 			ApplicationContext context) {
 		if (!StringUtils.hasText(properties.getInstanceId())) {
-			return normalizeForDns(
-					IdUtils.getDefaultInstanceId(context.getEnvironment(), false));
+			return IdUtils.getDefaultInstanceId(context.getEnvironment(),
```

Comment:
You removed `normalizeForDns()`, please add it back and undo changes to tests.

Reply:
Oh, I removed `normalizeForDns()` intentionally. As comments in #487 and #331 , maybe this method should be rewrite. And consul don't provide dns query for InstanceId, so i think it is no need to `normalizeForDns` for instanceId.

Reply:
I reset some commits and get it back. Maybe improve later.